### PR TITLE
Build noodler tests only if Catch2 is installed

### PR DESCRIPTION
**Pull request by jurajsic**

### CODE REVIEW

File: src/test/noodler/CMakeLists.txt [hunk 1]
```diff
@@ -1,35 +1,37 @@
...
-z3_add_install_tactic_rule(${z3_test_deps})
-z3_add_memory_initializer_rule(${z3_test_deps})
-z3_add_gparams_register_modules_rule(${z3_test_deps})
-target_compile_definitions(test-noodler PRIVATE ${Z3_COMPONENT_CXX_DEFINES})
-target_compile_options(test-noodler PRIVATE ${Z3_COMPONENT_CXX_FLAGS} -Wno-unused -Wno-unused-function)
-target_link_libraries(test-noodler PRIVATE ${Z3_DEPENDENT_LIBS})
-target_include_directories(test-noodler PRIVATE ${Z3_COMPONENT_EXTRA_INCLUDE_DIRS})
-z3_append_linker_flag_list_to_target(test-noodler ${Z3_DEPENDENT_EXTRA_CXX_LINK_FLAGS})
-z3_add_component_dependencies_to_target(test-noodler ${z3_test_expanded_deps})
-target_link_libraries(test-noodler PRIVATE Catch2::Catch2WithMain)
```
Comment (Adda0) [inline src/test/noodler/CMakeLists.txt:35]: Just a side-note: This `Catch2::Catch2WithMain` is how I would prefer to link Catch2 in Mata as well.

**Review by vhavlena — APPROVED** [review]
Great! Should we also write a short text to the README file how to install the Catch2? As far as I remember there were some problems during the installation.

**Comment by Adda0** [comment]
> Z3 still has 3.4 as its minimum, do we want to diverge here?

If we do not do it, Z3 will not compile soon-ish on many machines, including mine (with the newest CMake).

**Comment by Adda0** [comment]
> Maybe we should make a github action for tests, like it is done for mata, but Z3 takes so long to compile.

It can be done, but maybe only as a manually run action? Because compiling the whole of Z3 with each commit takes a looong time.

**Comment by Adda0** [comment]
> > Great! Should we also write a short text to the README file how to install the Catch2? As far as I remember there were some problems during the installation.
> 
> You mean for us? I think installing version 3 of Catch2 means cloning the repo and runnning cmake + make install, should I add that?

Yeah, I would add the comment. It will save some time one of each and others in the future.

**Comment by vhavlena** [comment]
> > Great! Should we also write a short text to the README file how to install the Catch2? As far as I remember there were some problems during the installation.
> 
> You mean for us? I think installing version 3 of Catch2 means cloning the repo and runnning cmake + make install, should I add that?

Yes, particularly for us and students. Maybe I am wrong, but the installation was not that easy. Did we need to set some system c++ variable (`CPLUS_INCLUDE_PATH`)? I don't remember correctly.

**Comment by vhavlena** [comment]
It works well on a system without `Catch2`.

**Comment by jurajsic** [comment]
> > > Great! Should we also write a short text to the README file how to install the Catch2? As far as I remember there were some problems during the installation.
> > 
> > 
> > You mean for us? I think installing version 3 of Catch2 means cloning the repo and runnning cmake + make install, should I add that?
> 
> Yes, particularly for us and students. Maybe I am wrong, but the installation was not that easy. Did we need to set some system c++ variable (`CPLUS_INCLUDE_PATH`)? I don't remember correctly.

I think `CPLUS_INCLUDE_PATH` was needed to set on mac for mata not for Catch2. If it is installed using brew, there should not be a problem.

**Comment by vhavlena** [comment]
> Tests should compile now, I also fixed some of them. Some tests for preprocessing fail, maybe @vhavlena can look at them to see if they are just testing something that should not be tested anymore, or if there is really some bug.

It should be fixed now.

**Comment by vhavlena** [comment]
Do we want to make some updates? Or is it ready for merge?

**Comment by jurajsic** [comment]
I am ok with merging it as is.

**Comment by Adda0** [comment]
I am happy with the changes, too.

**Pull request merged**